atomics/src/Lock_Array_SYCL.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+
+// FIXME_SYCL Use SYCL_EXT_ONEAPI_DEVICE_GLOBAL when available instead
+#ifdef DESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
+
+#include <cinttypes>
+#include <desul/atomics/Lock_Array_SYCL.hpp>
+#include <sstream>
+#include <string>
+
+namespace desul {
+namespace Impl {
+SYCL_EXTERNAL
+sycl::ext::oneapi::experimental::device_global<
+    int32_t*,
+    decltype(sycl::ext::oneapi::experimental::properties(
+        sycl::ext::oneapi::experimental::device_image_scope))>
+    SYCL_SPACE_ATOMIC_LOCKS_DEVICE;
+SYCL_EXTERNAL
+sycl::ext::oneapi::experimental::device_global<
+    int32_t*,
+    decltype(sycl::ext::oneapi::experimental::properties(
+        sycl::ext::oneapi::experimental::device_image_scope))>
+    SYCL_SPACE_ATOMIC_LOCKS_NODE;
+}  // namespace Impl
+}  // namespace desul
+
+namespace desul {
+namespace Impl {
+
+int32_t* SYCL_SPACE_ATOMIC_LOCKS_DEVICE_h = nullptr;
+int32_t* SYCL_SPACE_ATOMIC_LOCKS_NODE_h = nullptr;
+
+template <typename T>
+void init_lock_arrays_sycl() {
+  if (SYCL_SPACE_ATOMIC_LOCKS_DEVICE_h != nullptr) return;
+
+  sycl::queue q;
```

Comment:
How do we know that this queue is associated with the right device?

Reply:
Good question! It's probably better to just pass the `sycl::queue` as an argument.